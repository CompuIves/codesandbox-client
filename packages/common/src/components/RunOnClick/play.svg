<svg width="29" height="32" xmlns="http://www.w3.org/2000/svg" style="background:#ffd498"><g fill="none" fill-rule="evenodd"><path stroke="#FFF" stroke-width="3" fill="#6BAFDD" d="M2 23.023l12.456 6.954L26.916 23V9.017L14.458 2 2 9.048z"/><path d="M11.176 12.509c0-1.107.78-1.558 1.733-1.015l6.481 3.698c.957.546.952 1.434 0 1.978l-6.481 3.698c-.957.546-1.733.098-1.733-1.014v-7.345z" fill="#FFF"/></g></svg>