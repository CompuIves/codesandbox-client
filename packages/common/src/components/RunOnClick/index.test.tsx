import 'jest-styled-components';
import React from 'react';
import mountWithTheme from '../../test/themeMount';
import RunOnClick from './';

describe('<RunOnClick /> rendering', () => {
  it('basic', () => {
    const wrapper = mountWithTheme(<RunOnClick onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
