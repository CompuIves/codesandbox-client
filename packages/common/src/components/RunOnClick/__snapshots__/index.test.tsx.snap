// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RunOnClick /> rendering basic 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  width: 100%;
  height: 100%;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<RunOnClick
  onClick={[Function]}
>
  <styled.div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#FFD399",
        "cursor": "pointer",
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex: auto;
  width: 100%;
  height: 100%;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#FFD399",
          "cursor": "pointer",
        }
      }
    >
      <div
        className="c0"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#FFD399",
            "cursor": "pointer",
          }
        }
      >
        <styled.div
          horizontal={true}
          vertical={true}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  position: relative;
  display: flex;
  ",
                    [Function],
                    ";
  ",
                    [Function],
                    ";

  flex-direction: column;
  width: 100%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            horizontal={true}
            vertical={true}
          >
            <div
              className="c1"
            >
              <img
                alt="Run Sandbox"
                height={170}
                src={[Function]}
                width={170}
              />
              <div
                style={
                  Object {
                    "color": "#F27777",
                    "fontSize": "2rem",
                    "fontWeight": 700,
                    "marginTop": 24,
                    "textTransform": "uppercase",
                  }
                }
              >
                Click to run
              </div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</RunOnClick>
`;
